client/src/connection/itc/index.ts

```diff
@@ -0,0 +1,438 @@
+        this._runReject = undefined;
+        this._runResolve = undefined;
+      }
+      this.clearPassword();
+      resolve();
+      updateStatusBarItem(false);
+    });
+  };
+
+  /**
+   * Cancels a running SAS program
+   */
+  public cancel = async () => {
+    this._pollingForLogResults = false;
+    this._shellProcess.stdin.write("$runner.Cancel()\n", async (error) => {
+      if (error) {
+        this._runReject(error);
+      }
+
+      await this.fetchLog();
+    });
+  };
+
+  /**
+   * Formats the SAS Options provided in the profile into a format
+   * that the shell process can understand.
+   * @param sasOptions SAS Options array from the connection profile.
+   * @returns a string  denoting powershell syntax for an array literal.
+   */
+  private formatSASOptions = (sasOptions: string[]): string => {
+    const optionsVariable = `@("${sasOptions.join(`","`)}")`;
+    return optionsVariable;
+  };
+
+  /**
+   * Flushes the SAS log in chunks of [chunkSize] length,
+   * writing each chunk to stdout.
+   */
+  private fetchLog = async (): Promise<void> => {
+    const pollingInterval = setInterval(() => {
+      if (!this._pollingForLogResults) {
+        clearInterval(pollingInterval);
+      }
+      this._shellProcess.stdin.write(
+        `
+  do {
+    $chunkSize = 32768
+    $log = $runner.FlushLog($chunkSize)
+    Write-Host $log
+  } while ($log.Length -gt 0)\n
+    `,
+        this.onWriteComplete,
+      );
+    }, 2 * 1000);
+  };
+
+  /**
+   * Handles stderr output from the powershell child process.
+   * @param chunk a buffer of stderr output from the child process.
+   */
+  private onShellStdErr = (chunk: Buffer): void => {
+    const msg = chunk.toString();
+    console.warn("shellProcess stderr: " + msg);
+    this.clearPassword();
+    this._runReject(
+      new Error(
+        "There was an error executing the SAS Program.\nSee console log for more details.",
+      ),
+    );
+    // If we encountered an error in setup, we need to go through everything again
+    if (/Setup error/.test(msg)) {
+      this._shellProcess.kill();
+      this._workDirectory = undefined;
+    }
+  };
+
+  /**
+   * Handles stdout output from the powershell child process.
+   * @param data a buffer of stdout output from the child process.
+   */
+  private onShellStdOut = (data: Buffer): void => {
+    const output = data.toString().trimEnd();
+    const outputLines = output.split(/\n|\r\n/);
+
+    outputLines.forEach((line: string) => {
+      if (!line) {
+        return;
+      }
+      if (!this._workDirectory && line.startsWith("WORKDIR=")) {
+        const parts = line.split("WORKDIR=");
+        this._workDirectory = parts[1].trim();
+        this._runResolve();
+        updateStatusBarItem(true);
+        return;
+      }
+      if (!this.processLineCodes(line)) {
+        this._html5FileName =
+          line.match(/NOTE: .+ HTML5.* Body.+: (.+)\.htm/)?.[1] ??
+          this._html5FileName;
```

Comment:
I think it should reset `this._html5FileName` upon each run, as ssh does
https://github.com/sassoftware/vscode-sas-extension/blob/main/client/src/connection/ssh/index.ts#L89
Otherwise it will use the name from last run if no new ods result produced

Reply:
Resolved in https://github.com/sassoftware/vscode-sas-extension/pull/592/commits/3b80433ddddc02f46f6b00c2e633547af8fdf477